lib/qdrant-docker-image-ecr-kms-deployment-cdk-stack.ts

```diff
@@ -0,0 +1,34 @@
+import * as kms from 'aws-cdk-lib/aws-kms';
+import * as cdk from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as ecrDeploy from 'cdk-ecr-deployment';
+import * as ecr from 'aws-cdk-lib/aws-ecr';
+import { QdrantDockerImageEcrDeploymentCdkStackProps } from './QdrantDockerImageEcrDeploymentCdkStackProps';
+
+export class QdrantDockerImageEcrDeploymentCdkStack extends cdk.Stack {
+    constructor(scope: Construct, id: string, props: QdrantDockerImageEcrDeploymentCdkStackProps) {
+        super(scope, id, props);
+
+        const kmsKey = new kms.Key(this, `${props.appName}-ECRRepositoryKmsKey`, {
+            enableKeyRotation: true,
+            removalPolicy: cdk.RemovalPolicy.DESTROY,
+            enabled: true,
+        });
+
+        const ecrRepository = new ecr.Repository(this, `${props.appName}-QdrantDockerImageEcrRepository`, {
+            repositoryName: props?.repositoryName ?? 'qdrant-docker-image-ecr-kms-deployment-cdk',
```

Comment:
**Suggestion:** The `repositoryName` property in the `ecr.Repository` constructor should not have a fallback since it is already defined as a required property in `QdrantDockerImageEcrDeploymentCdkStackProps`.
```suggestion
        const ecrRepository = new ecr.Repository(this, `${props.appName}-QdrantDockerImageEcrRepository`, {
            repositoryName: props.repositoryName,
            ...
        });
```